src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/SmoModel/SmoChildFactoryBase.cs

```diff
@@ -50,6 +46,28 @@ public override IEnumerable<TreeNode> Expand(TreeNode parent, bool refresh, stri
             return allChildren;
         }
 
+        private void OnExpandPopulateFoldersAndFilter(List<TreeNode> allChildren, TreeNode parent, bool includeSystemObjects)
+        {
+            SmoQueryContext context = parent.GetContextAs<SmoQueryContext>();
+            OnExpandPopulateFolders(allChildren, parent);
+            if (!includeSystemObjects)
+            {
+                allChildren.RemoveAll(x => x.IsSystemObject);
+            }
+            if (context != null && context.ValidFor != 0 && context.ValidFor != ValidForFlag.All)

```

Comment:
Minor: It's nicer if we define a `ValidForFlag.None` as 0 and then test against that